<?xml version="1.0" ?>
<!-- =================================================================================== -->
<!-- |    This document was autogenerated by xacro from spm.urdf.xacro                 | -->
<!-- |    EDITING THIS FILE BY HAND IS NOT RECOMMENDED                                 | -->
<!-- =================================================================================== -->
<robot name="spm_wrist">
  <material name="silver_default">
    <color rgba="0.700 0.700 0.700 1.000"/>
  </material>
  <material name="epx_150_air_baked_with_carbon_m2_m3_l1_3d_printers">
    <color rgba="0.24705882352941178 0.24705882352941178 0.24705882352941178 1.0"/>
  </material>
  <material name="nylon_12_with_formlabs_fuse_1_3d_printer">
    <color rgba="0.24705882352941178 0.24705882352941178 0.24705882352941178 1.0"/>
  </material>
  <material name="steel_satin">
    <color rgba="0.6274509803921569 0.6274509803921569 0.6274509803921569 1.0"/>
  </material>
  <material name="printed_polymide_pa12cf">
    <color rgba="0.24705882352941178 0.24705882352941178 0.24705882352941178 1.0"/>
  </material>
  <link name="base_link">
    <inertial>
      <origin rpy="0 0 0" xyz="1.4144943148930138e-06 0.007917232444087631 -6.354750979618071e-06"/>
      <mass value="0.046353737838188414"/>
      <inertia ixx="2.9e-05" ixy="0.0" ixz="0.0" iyy="4.2e-05" iyz="-0.0" izz="2.9e-05"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://spm_wrist_description/meshes/base_link.stl" scale="0.001 0.001 0.001"/>
      </geometry>
      <material name="epx_150_air_baked_with_carbon_m2_m3_l1_3d_printers"/>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://spm_wrist_description/meshes/base_link.stl" scale="0.001 0.001 0.001"/>
      </geometry>
    </collision>
  </link>
  <link name="ServoMotor1">
    <inertial>
      <origin rpy="0 0 0" xyz="0.011948473139161964 0.016078734989187438 -0.018863335082905573"/>
      <mass value="0.039200496616158965"/>
      <inertia ixx="6e-06" ixy="-0.0" ixz="2e-06" iyy="9e-06" iyz="0.0" izz="9e-06"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0.039051 -0.0015 -0.034511"/>
      <geometry>
        <mesh filename="package://spm_wrist_description/meshes/servo_motor_v2_1.stl" scale="0.001 0.001 0.001"/>
      </geometry>
      <material name="nylon_12_with_formlabs_fuse_1_3d_printer"/>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0.039051 -0.0015 -0.034511"/>
      <geometry>
        <mesh filename="package://spm_wrist_description/meshes/servo_motor_v2_1.stl" scale="0.001 0.001 0.001"/>
      </geometry>
    </collision>
  </link>
  <link name="ServoMotor2">
    <inertial>
      <origin rpy="0 0 0" xyz="-0.022309461241470315 0.016078734989208664 -0.0009163556903109618"/>
      <mass value="0.039200496616158965"/>
      <inertia ixx="6e-06" ixy="0.0" ixz="-2e-06" iyy="9e-06" iyz="0.0" izz="9e-06"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="-0.049412 -0.0015 -0.016564"/>
      <geometry>
        <mesh filename="package://spm_wrist_description/meshes/servo_motor_v2_2.stl" scale="0.001 0.001 0.001"/>
      </geometry>
      <material name="nylon_12_with_formlabs_fuse_1_3d_printer"/>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="-0.049412 -0.0015 -0.016564"/>
      <geometry>
        <mesh filename="package://spm_wrist_description/meshes/servo_motor_v2_2.stl" scale="0.001 0.001 0.001"/>
      </geometry>
    </collision>
  </link>
  <link name="ServoMotor3">
    <inertial>
      <origin rpy="0 0 0" xyz="0.010361988102308841 0.0160787349892087 0.019778690773216326"/>
      <mass value="0.039200496616158965"/>
      <inertia ixx="1e-05" ixy="-0.0" ixz="-0.0" iyy="9e-06" iyz="-0.0" izz="5e-06"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0.010362 -0.0015 0.051074"/>
      <geometry>
        <mesh filename="package://spm_wrist_description/meshes/servo_motor_v2_3.stl" scale="0.001 0.001 0.001"/>
      </geometry>
      <material name="nylon_12_with_formlabs_fuse_1_3d_printer"/>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0.010362 -0.0015 0.051074"/>
      <geometry>
        <mesh filename="package://spm_wrist_description/meshes/servo_motor_v2_3.stl" scale="0.001 0.001 0.001"/>
      </geometry>
    </collision>
  </link>
  <link name="AxisConnector1">
    <inertial>
      <origin rpy="0 0 0" xyz="1.6013453995114557e-08 0.0024852853726477706 2.515422694148589e-07"/>
      <mass value="0.004090743480523257"/>
      <inertia ixx="8.4649e-08" ixy="0.0" ixz="0.0" iyy="1.5336e-07" iyz="0.0" izz="8.4649e-08"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0.015556 -0.0352 -0.008981"/>
      <geometry>
        <mesh filename="package://spm_wrist_description/meshes/AxisConnector_v3_1.stl" scale="0.001 0.001 0.001"/>
      </geometry>
      <material name="steel_satin"/>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0.015556 -0.0352 -0.008981"/>
      <geometry>
        <mesh filename="package://spm_wrist_description/meshes/AxisConnector_v3_1.stl" scale="0.001 0.001 0.001"/>
      </geometry>
    </collision>
  </link>
  <link name="AxisConnector2">
    <inertial>
      <origin rpy="0 0 0" xyz="-1.6013119337465742e-08 0.0024852853726689966 2.5154246256937907e-07"/>
      <mass value="0.004090743480523257"/>
      <inertia ixx="8.4649e-08" ixy="0.0" ixz="0.0" iyy="1.5336e-07" iyz="0.0" izz="8.4649e-08"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="-0.015556 -0.0352 -0.008981"/>
      <geometry>
        <mesh filename="package://spm_wrist_description/meshes/AxisConnector_v3_2.stl" scale="0.001 0.001 0.001"/>
      </geometry>
      <material name="steel_satin"/>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="-0.015556 -0.0352 -0.008981"/>
      <geometry>
        <mesh filename="package://spm_wrist_description/meshes/AxisConnector_v3_2.stl" scale="0.001 0.001 0.001"/>
      </geometry>
    </collision>
  </link>
  <link name="AxisConnector3">
    <inertial>
      <origin rpy="0 0 0" xyz="-1.0994205545955537e-13 0.0024852853726690244 4.969156521712759e-07"/>
      <mass value="0.004090743480523257"/>
      <inertia ixx="8.4649e-08" ixy="0.0" ixz="0.0" iyy="1.5336e-07" iyz="0.0" izz="8.4649e-08"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="-0.0 -0.0352 0.017963"/>
      <geometry>
        <mesh filename="package://spm_wrist_description/meshes/AxisConnector_v3_3.stl" scale="0.001 0.001 0.001"/>
      </geometry>
      <material name="steel_satin"/>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="-0.0 -0.0352 0.017963"/>
      <geometry>
        <mesh filename="package://spm_wrist_description/meshes/AxisConnector_v3_3.stl" scale="0.001 0.001 0.001"/>
      </geometry>
    </collision>
  </link>
  <link name="LngSpur">
    <inertial>
      <origin rpy="0 0 0" xyz="1.6013355952013142e-08 0.023811024494842743 2.515424875615402e-07"/>
      <mass value="0.002241786298874095"/>
      <inertia ixx="0.0" ixy="-0.0" ixz="-0.0" iyy="0.0" iyz="-0.0" izz="0.0"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0.015556 -0.0382 -0.008981"/>
      <geometry>
        <mesh filename="package://spm_wrist_description/meshes/LngSpur_v2_1.stl" scale="0.001 0.001 0.001"/>
      </geometry>
      <material name="epx_150_air_baked_with_carbon_m2_m3_l1_3d_printers"/>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0.015556 -0.0382 -0.008981"/>
      <geometry>
        <mesh filename="package://spm_wrist_description/meshes/LngSpur_v2_1.stl" scale="0.001 0.001 0.001"/>
      </geometry>
    </collision>
  </link>
  <link name="MidSpur">
    <inertial>
      <origin rpy="0 0 0" xyz="-0.003099851187174882 0.021909865513360466 -5.61942304570795e-07"/>
      <mass value="0.0022422400558409607"/>
      <inertia ixx="0.0" ixy="-0.0" ixz="0.0" iyy="0.0" iyz="0.0" izz="0.0"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="-0.003 -0.0382 0.017963"/>
      <geometry>
        <mesh filename="package://spm_wrist_description/meshes/MidSpur_v2_1.stl" scale="0.001 0.001 0.001"/>
      </geometry>
      <material name="epx_150_air_baked_with_carbon_m2_m3_l1_3d_printers"/>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="-0.003 -0.0382 0.017963"/>
      <geometry>
        <mesh filename="package://spm_wrist_description/meshes/MidSpur_v2_1.stl" scale="0.001 0.001 0.001"/>
      </geometry>
    </collision>
  </link>
  <link name="SrtSpur">
    <inertial>
      <origin rpy="0 0 0" xyz="-0.0031000160130173603 0.020004935604028537 2.515421796325107e-07"/>
      <mass value="0.0022417862988739273"/>
      <inertia ixx="0.0" ixy="0.0" ixz="0.0" iyy="0.0" iyz="0.0" izz="0.0"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="-0.018556 -0.0382 -0.008981"/>
      <geometry>
        <mesh filename="package://spm_wrist_description/meshes/SrtSpur_v2_1.stl" scale="0.001 0.001 0.001"/>
      </geometry>
      <material name="epx_150_air_baked_with_carbon_m2_m3_l1_3d_printers"/>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="-0.018556 -0.0382 -0.008981"/>
      <geometry>
        <mesh filename="package://spm_wrist_description/meshes/SrtSpur_v2_1.stl" scale="0.001 0.001 0.001"/>
      </geometry>
    </collision>
  </link>
  <link name="Bottom1">
    <inertial>
      <origin rpy="0 0 0" xyz="1.289708778125881e-08 0.006279746707014873 -5.8722048078617854e-09"/>
      <mass value="0.016848685557919317"/>
      <inertia ixx="3e-06" ixy="0.0" ixz="-0.0" iyy="6e-06" iyz="-0.0" izz="3e-06"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="-0.0 -0.0385 -0.0"/>
      <geometry>
        <mesh filename="package://spm_wrist_description/meshes/Bottom_v2_1.stl" scale="0.001 0.001 0.001"/>
      </geometry>
      <material name="epx_150_air_baked_with_carbon_m2_m3_l1_3d_printers"/>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="-0.0 -0.0385 -0.0"/>
      <geometry>
        <mesh filename="package://spm_wrist_description/meshes/Bottom_v2_1.stl" scale="0.001 0.001 0.001"/>
      </geometry>
    </collision>
  </link>
  <link name="Bottom2">
    <inertial>
      <origin rpy="0 0 0" xyz="1.289708778125881e-08 0.022279746707014866 -5.8722048078617854e-09"/>
      <mass value="0.016848685557919317"/>
      <inertia ixx="3e-06" ixy="0.0" ixz="-0.0" iyy="6e-06" iyz="-0.0" izz="3e-06"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="-0.0 -0.0385 -0.0"/>
      <geometry>
        <mesh filename="package://spm_wrist_description/meshes/Bottom_v2_2.stl" scale="0.001 0.001 0.001"/>
      </geometry>
      <material name="epx_150_air_baked_with_carbon_m2_m3_l1_3d_printers"/>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="-0.0 -0.0385 -0.0"/>
      <geometry>
        <mesh filename="package://spm_wrist_description/meshes/Bottom_v2_2.stl" scale="0.001 0.001 0.001"/>
      </geometry>
    </collision>
  </link>
  <link name="Bottom3">
    <inertial>
      <origin rpy="0 0 0" xyz="1.4260146814520231e-08 0.038279746707015776 -6.65916595122069e-09"/>
      <mass value="0.016848685557919317"/>
      <inertia ixx="3e-06" ixy="0.0" ixz="-0.0" iyy="6e-06" iyz="-0.0" izz="3e-06"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="-0.0 -0.0385 -0.0"/>
      <geometry>
        <mesh filename="package://spm_wrist_description/meshes/Bottom_v2_3.stl" scale="0.001 0.001 0.001"/>
      </geometry>
      <material name="epx_150_air_baked_with_carbon_m2_m3_l1_3d_printers"/>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="-0.0 -0.0385 -0.0"/>
      <geometry>
        <mesh filename="package://spm_wrist_description/meshes/Bottom_v2_3.stl" scale="0.001 0.001 0.001"/>
      </geometry>
    </collision>
  </link>
  <link name="Ring1">
    <inertial>
      <origin rpy="0 0 0" xyz="0.005201501758024685 0.007609632754506879 4.973880009166959e-06"/>
      <mass value="0.010986111735110429"/>
      <inertia ixx="4e-06" ixy="0.0" ixz="-0.0" iyy="9e-06" iyz="0.0" izz="5e-06"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0.0 -0.0405 -0.0"/>
      <geometry>
        <mesh filename="package://spm_wrist_description/meshes/Ring_v2_1.stl" scale="0.001 0.001 0.001"/>
      </geometry>
      <material name="epx_150_air_baked_with_carbon_m2_m3_l1_3d_printers"/>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0.0 -0.0405 -0.0"/>
      <geometry>
        <mesh filename="package://spm_wrist_description/meshes/Ring_v2_1.stl" scale="0.001 0.001 0.001"/>
      </geometry>
    </collision>
  </link>
  <link name="Ring2">
    <inertial>
      <origin rpy="0 0 0" xyz="-0.0025964447356298537 0.007609632754506872 -0.004507117239401916"/>
      <mass value="0.010986111735110429"/>
      <inertia ixx="5e-06" ixy="-0.0" ixz="-1e-06" iyy="9e-06" iyz="-0.0" izz="4e-06"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0.0 -0.0565 -0.0"/>
      <geometry>
        <mesh filename="package://spm_wrist_description/meshes/Ring_v2_2.stl" scale="0.001 0.001 0.001"/>
      </geometry>
      <material name="epx_150_air_baked_with_carbon_m2_m3_l1_3d_printers"/>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0.0 -0.0565 -0.0"/>
      <geometry>
        <mesh filename="package://spm_wrist_description/meshes/Ring_v2_2.stl" scale="0.001 0.001 0.001"/>
      </geometry>
    </collision>
  </link>
  <link name="Ring3">
    <inertial>
      <origin rpy="0 0 0" xyz="-0.0026050556593440444 0.007609632754507795 0.0045021472942029135"/>
      <mass value="0.010986111735110429"/>
      <inertia ixx="5e-06" ixy="-0.0" ixz="1e-06" iyy="9e-06" iyz="0.0" izz="4e-06"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="-0.0 -0.0725 -0.0"/>
      <geometry>
        <mesh filename="package://spm_wrist_description/meshes/Ring_v2_3.stl" scale="0.001 0.001 0.001"/>
      </geometry>
      <material name="epx_150_air_baked_with_carbon_m2_m3_l1_3d_printers"/>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="-0.0 -0.0725 -0.0"/>
      <geometry>
        <mesh filename="package://spm_wrist_description/meshes/Ring_v2_3.stl" scale="0.001 0.001 0.001"/>
      </geometry>
    </collision>
  </link>
  <link name="LongArm">
    <inertial>
      <origin rpy="0 0 0" xyz="0.00038460246617244565 0.02537308365506035 0.00500888165590829"/>
      <mass value="0.0051004386314811665"/>
      <inertia ixx="1e-06" ixy="-0.0" ixz="0.0" iyy="0.0" iyz="0.0" izz="1e-06"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="-0.03431 -0.053 0.00499"/>
      <geometry>
        <mesh filename="package://spm_wrist_description/meshes/LongArm_v2_1.stl" scale="0.001 0.001 0.001"/>
      </geometry>
      <material name="printed_polymide_pa12cf"/>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="-0.03431 -0.053 0.00499"/>
      <geometry>
        <mesh filename="package://spm_wrist_description/meshes/LongArm_v2_1.stl" scale="0.001 0.001 0.001"/>
      </geometry>
    </collision>
  </link>
  <link name="MiddleArm">
    <inertial>
      <origin rpy="0 0 0" xyz="0.0040539394551962205 0.01736881870394401 -0.0029924213263613428"/>
      <mass value="0.003478285860946649"/>
      <inertia ixx="0.0" ixy="0.0" ixz="0.0" iyy="0.0" iyz="0.0" izz="0.0"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0.021476 -0.069 0.027218"/>
      <geometry>
        <mesh filename="package://spm_wrist_description/meshes/MiddleArm_v2_1.stl" scale="0.001 0.001 0.001"/>
      </geometry>
      <material name="epx_150_air_baked_with_carbon_m2_m3_l1_3d_printers"/>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0.021476 -0.069 0.027218"/>
      <geometry>
        <mesh filename="package://spm_wrist_description/meshes/MiddleArm_v2_1.stl" scale="0.001 0.001 0.001"/>
      </geometry>
    </collision>
  </link>
  <link name="ShortArm">
    <inertial>
      <origin rpy="0 0 0" xyz="0.003816843793053231 0.009294845203876759 0.003387634092917313"/>
      <mass value="0.001981898490704533"/>
      <inertia ixx="0.0" ixy="0.0" ixz="-0.0" iyy="0.0" iyz="-0.0" izz="0.0"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0.021494 -0.085 -0.027208"/>
      <geometry>
        <mesh filename="package://spm_wrist_description/meshes/ShortArm_v2_1.stl" scale="0.001 0.001 0.001"/>
      </geometry>
      <material name="epx_150_air_baked_with_carbon_m2_m3_l1_3d_printers"/>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0.021494 -0.085 -0.027208"/>
      <geometry>
        <mesh filename="package://spm_wrist_description/meshes/ShortArm_v2_1.stl" scale="0.001 0.001 0.001"/>
      </geometry>
    </collision>
  </link>
  <link name="PlatformSupportLink1">
    <inertial>
      <origin rpy="0 0 0" xyz="-0.00818592553194598 0.02354354691894711 0.01714252319173745"/>
      <mass value="0.010151333245456888"/>
      <inertia ixx="3e-06" ixy="1e-06" ixz="1e-06" iyy="3e-06" iyz="-1e-06" izz="4e-06"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="-0.031815 -0.101598 -0.0"/>
      <geometry>
        <mesh filename="package://spm_wrist_description/meshes/PlatformSupportLink_v2_1.stl" scale="0.001 0.001 0.001"/>
      </geometry>
      <material name="epx_150_air_baked_with_carbon_m2_m3_l1_3d_printers"/>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="-0.031815 -0.101598 -0.0"/>
      <geometry>
        <mesh filename="package://spm_wrist_description/meshes/PlatformSupportLink_v2_1.stl" scale="0.001 0.001 0.001"/>
      </geometry>
    </collision>
  </link>
  <link name="PlatformSupportLink2">
    <inertial>
      <origin rpy="0 0 0" xyz="0.019526903785241048 0.023070907887856165 -0.0012751235405861032"/>
      <mass value="0.010151333245456888"/>
      <inertia ixx="2.757e-06" ixy="-1.541e-06" ixz="-8.53e-07" iyy="3.533e-06" iyz="1.0e-08" izz="4.447e-06"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0.015908 -0.101598 0.027553"/>
      <geometry>
        <mesh filename="package://spm_wrist_description/meshes/PlatformSupportLink_v2_2.stl" scale="0.001 0.001 0.001"/>
      </geometry>
      <material name="epx_150_air_baked_with_carbon_m2_m3_l1_3d_printers"/>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0.015908 -0.101598 0.027553"/>
      <geometry>
        <mesh filename="package://spm_wrist_description/meshes/PlatformSupportLink_v2_2.stl" scale="0.001 0.001 0.001"/>
      </geometry>
    </collision>
  </link>
  <link name="PlatformSupportLink3">
    <inertial>
      <origin rpy="0 0 0" xyz="-0.010893843297945977 0.022943909773968235 -0.016434793066780543"/>
      <mass value="0.010151333245456888"/>
      <inertia ixx="4.758e-06" ixy="7.7e-07" ixz="-2.99e-07" iyy="3.552e-06" iyz="1.33e-06" izz="2.427e-06"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0.015906 -0.101598 -0.027554"/>
      <geometry>
        <mesh filename="package://spm_wrist_description/meshes/PlatformSupportLink_v2_3.stl" scale="0.001 0.001 0.001"/>
      </geometry>
      <material name="epx_150_air_baked_with_carbon_m2_m3_l1_3d_printers"/>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0.015906 -0.101598 -0.027554"/>
      <geometry>
        <mesh filename="package://spm_wrist_description/meshes/PlatformSupportLink_v2_3.stl" scale="0.001 0.001 0.001"/>
      </geometry>
    </collision>
  </link>
  <link name="Platform">
    <inertial>
      <origin rpy="0 0 0" xyz="-0.0007498067553170075 0.0014759244320091403 -0.0275398582616901"/>
      <mass value="0.008694089397605359"/>
      <inertia ixx="2e-06" ixy="-0.0" ixz="0.0" iyy="3e-06" iyz="-0.0" izz="2e-06"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="-0.000885 -0.134013 -0.02748"/>
      <geometry>
        <mesh filename="package://spm_wrist_description/meshes/Platform_v2_1.stl" scale="0.001 0.001 0.001"/>
      </geometry>
      <material name="epx_150_air_baked_with_carbon_m2_m3_l1_3d_printers"/>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="-0.000885 -0.134013 -0.02748"/>
      <geometry>
        <mesh filename="package://spm_wrist_description/meshes/Platform_v2_1.stl" scale="0.001 0.001 0.001"/>
      </geometry>
    </collision>
  </link>
  <link name="Top">
    <inertial>
      <origin rpy="0 0 0" xyz="1.3631331504828307e-09 0.0010000000000000425 -0.016999999213027802"/>
      <mass value="0.004660046355404222"/>
      <inertia ixx="1e-06" ixy="-0.0" ixz="0.0" iyy="2e-06" iyz="-0.0" izz="1e-06"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="-0.0 -0.0865 -0.017"/>
      <geometry>
        <mesh filename="package://spm_wrist_description/meshes/Top_v2_1.stl" scale="0.001 0.001 0.001"/>
      </geometry>
      <material name="epx_150_air_baked_with_carbon_m2_m3_l1_3d_printers"/>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="-0.0 -0.0865 -0.017"/>
      <geometry>
        <mesh filename="package://spm_wrist_description/meshes/Top_v2_1.stl" scale="0.001 0.001 0.001"/>
      </geometry>
    </collision>
  </link>
  <joint name="Rigid1" type="fixed">
    <origin rpy="0 0 0" xyz="-0.039051 0.0015 0.034511"/>
    <parent link="base_link"/>
    <child link="ServoMotor1"/>
  </joint>
  <joint name="Rigid2" type="fixed">
    <origin rpy="0 0 0" xyz="0.049412 0.0015 0.016564"/>
    <parent link="base_link"/>
    <child link="ServoMotor2"/>
  </joint>
  <joint name="Rigid3" type="fixed">
    <origin rpy="0 0 0" xyz="-0.010362 0.0015 -0.051074"/>
    <parent link="base_link"/>
    <child link="ServoMotor3"/>
  </joint>
  <joint name="LngSpur_Joint" type="continuous">
    <origin rpy="0 0 0" xyz="0.023495 0.0337 -0.02553"/>
    <parent link="ServoMotor1"/>
    <child link="AxisConnector1"/>
    <axis xyz="-0.0 -1.0 -0.0"/>
  </joint>
  <joint name="SrtSpur_Joint" type="continuous">
    <origin rpy="0 0 0" xyz="-0.033856 0.0337 -0.007583"/>
    <parent link="ServoMotor2"/>
    <child link="AxisConnector2"/>
    <axis xyz="-0.0 1.0 0.0"/>
  </joint>
  <joint name="MidSpur_Joint" type="continuous">
    <origin rpy="0 0 0" xyz="0.010362 0.0337 0.033111"/>
    <parent link="ServoMotor3"/>
    <child link="AxisConnector3"/>
    <axis xyz="0.0 1.0 0.0"/>
  </joint>
  <joint name="Rigid7" type="fixed">
    <origin rpy="0 0 0" xyz="0.0 0.003 0.0"/>
    <parent link="AxisConnector1"/>
    <child link="LngSpur"/>
  </joint>
  <joint name="Rigid8" type="fixed">
    <origin rpy="0 0 0" xyz="0.003 0.003 0.0"/>
    <parent link="AxisConnector3"/>
    <child link="MidSpur"/>
  </joint>
  <joint name="Rigid9" type="fixed">
    <origin rpy="0 0 0" xyz="0.003 0.003 0.0"/>
    <parent link="AxisConnector2"/>
    <child link="SrtSpur"/>
  </joint>
  <joint name="Rigid10" type="fixed">
    <origin rpy="0 0 0" xyz="0.0 0.0385 0.0"/>
    <parent link="base_link"/>
    <child link="Bottom1"/>
  </joint>
  <joint name="Rigid11" type="fixed">
    <origin rpy="0 0 0" xyz="0.0 0.0385 0.0"/>
    <parent link="base_link"/>
    <child link="Bottom2"/>
  </joint>
  <joint name="Rigid12" type="fixed">
    <origin rpy="0 0 0" xyz="0.0 0.0385 0.0"/>
    <parent link="base_link"/>
    <child link="Bottom3"/>
  </joint>
  <joint name="Revolute13" type="continuous">
    <origin rpy="0 0 0" xyz="-0.0 0.002 0.0"/>
    <parent link="Bottom1"/>
    <child link="Ring1"/>
    <axis xyz="-0.0 1.0 -0.0"/>
    <mimic joint="LngSpur_Joint" multiplier="-0.4"/>
  </joint>
  <joint name="Revolute14" type="continuous">
    <origin rpy="0 0 0" xyz="-0.0 0.018 0.0"/>
    <parent link="Bottom2"/>
    <child link="Ring2"/>
    <axis xyz="0.0 1.0 0.0"/>
    <mimic joint="MidSpur_Joint" multiplier="-0.4"/>
  </joint>
  <joint name="Revolute15" type="continuous">
    <origin rpy="0 0 0" xyz="0.0 0.034 0.0"/>
    <parent link="Bottom3"/>
    <child link="Ring3"/>
    <axis xyz="0.0 1.0 0.0"/>
    <mimic joint="SrtSpur_Joint" multiplier="-0.4"/>
  </joint>
  <joint name="Rigid16" type="fixed">
    <origin rpy="0 0 0" xyz="0.03431 0.0125 -0.00499"/>
    <parent link="Ring1"/>
    <child link="LongArm"/>
  </joint>
  <joint name="Rigid17" type="fixed">
    <origin rpy="0 0 0" xyz="-0.021476 0.0125 -0.027218"/>
    <parent link="Ring2"/>
    <child link="MiddleArm"/>
  </joint>
  <joint name="Rigid18" type="fixed">
    <origin rpy="0 0 0" xyz="-0.021494 0.0125 0.027208"/>
    <parent link="Ring3"/>
    <child link="ShortArm"/>
  </joint>
  <joint name="PlatformSupportLink1_Joint" type="continuous">
    <origin rpy="0 0 0" xyz="-0.002495 0.048598 0.00499"/>
    <parent link="LongArm"/>
    <child link="PlatformSupportLink1"/>
    <axis xyz="-0.707107 0.707107 -0.0"/>
  </joint>
  <joint name="PlatformSupportLink2_Joint" type="continuous">
    <origin rpy="0 0 0" xyz="0.005568 0.032598 -0.000335"/>
    <parent link="MiddleArm"/>
    <child link="PlatformSupportLink2"/>
    <axis xyz="0.353553 0.707107 0.612372"/>
  </joint>
  <joint name="PlatformSupportLink3_Joint" type="continuous">
    <origin rpy="0 0 0" xyz="0.005588 0.016598 0.000346"/>
    <parent link="ShortArm"/>
    <child link="PlatformSupportLink3"/>
    <axis xyz="-0.353553 -0.707107 0.612372"/>
  </joint>
  <joint name="Platform_Joint1" type="continuous">
    <origin rpy="0 0 0" xyz="-0.03093 0.032415 0.02748"/>
    <parent link="PlatformSupportLink1"/>
    <child link="Platform"/>
    <axis xyz="-0.027239 -0.027239 -0.999258"/>
  </joint>
  <joint name="Rigid23" type="fixed">
    <origin rpy="0 0 0" xyz="0.0 0.048 0.017"/>
    <parent link="Bottom3"/>
    <child link="Top"/>
  </joint>
  <gazebo>
    <joint name="Platform_Joint2" type="revolute">
      <pose relative_to="PlatformSupportLink2">0.03879 0.03167 0.01419 0 -1.0471975500000001 0</pose>
      <parent>Platform</parent>
      <child>PlatformSupportLink2</child>
      <axis>
        <xyz> 0 0 1</xyz>
      </axis>
      <limit>
        <lower>-3.14159265</lower>
        <upper>3.14159265</upper>
        <effort>100.0</effort>
        <velocity>10.0</velocity>
      </limit>
      <dynamics>
        <damping>10.0</damping>
        <friction>1.0</friction>
      </dynamics>
    </joint>
    <joint name="Platform_Joint3" type="revolute">
      <pose relative_to="PlatformSupportLink3">-0.007013 0.03142 -0.04085 0 1.0471975500000001 0</pose>
      <parent>Platform</parent>
      <child>PlatformSupportLink3</child>
      <axis>
        <xyz> 0 0 1</xyz>
      </axis>
      <limit>
        <lower>-3.14159265</lower>
        <upper>3.14159265</upper>
        <effort>100.0</effort>
        <velocity>10.0</velocity>
      </limit>
      <dynamics>
        <damping>10.0</damping>
        <friction>1.0</friction>
      </dynamics>
    </joint>
  </gazebo>
  <link name="world"/>
  <joint name="world_joint" type="fixed">
    <parent link="world"/>
    <child link="base_link"/>
    <origin rpy="1.570796325 0.0 0.0" xyz="0.0 0.0 0.0"/>
  </joint>
</robot>
